src/c++/perf_analyzer/command_line_parser.cc

```diff
@@ -804,157 +834,215 @@ CLParser::ParseCommandLine(int argc, char** argv)
   while ((opt = getopt_long(
               argc, argv, "vdazc:u:m:x:b:t:p:i:H:l:r:s:f:", long_options,
               NULL)) != -1) {
-    switch (opt) {
-      case 0:
-        params_->streaming = true;
-        break;
-      case 1:
-        params_->max_threads = std::atoi(optarg);
-        params_->max_threads_specified = true;
-        break;
-      case 2:
-        params_->sequence_length = std::atoi(optarg);
-        params_->sequence_length_specified = true;
-        break;
-      case 3:
-        params_->percentile = std::atoi(optarg);
-        break;
-      case 4:
-        params_->user_data.push_back(optarg);
-        break;
-      case 5: {
-        std::string arg = optarg;
-        auto colon_pos = arg.rfind(":");
-        if (colon_pos == std::string::npos) {
-          Usage(
-              "failed to parse input shape. There must be a colon after input "
-              "name.");
+    try {
```

Comment:
I think I prefer to leave the try/catch for `std::invalid_argument` local to each case that previously checked it (there's only 4). That way there's less overall indentation, which imo makes the code less readable.

I think it's also a little clearer to the reader as to what the try catch is intending to catch without needing to scroll as far.

Reply:
The thing is there were only 4 before but we will need much more to cover every other flags (e.g. every flag that takes number as argument and therefore calls `std::sto*`). I can push the try/catch block up to `CLParser::Parse` function but in this case, it will be harder to spot the flag that's complaining.

I guess this is also where the custom functions like `IsNonNegativeNumber` could be useful. Instead of relying on `std::sto*` functions and one giant try/catch block to catch invalid arguments, we could directly check for them by parsing the inputs ourselves and throw exception and useful error messages when spotted.